CMakeLists.txt

```diff
@@ -70,6 +72,30 @@ option(BUILD_SLIMGUI "Build the Qt5-based GUI for SLiM" OFF)
 set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -std=c11 -Wno-attributes -Wunused-label -Wimplicit -Wunused-variable -Wunused-value -Wno-pragmas -Wempty-body -Wshadow -Wparentheses -Wmissing-prototypes -Wswitch -Wpointer-sign -Wsign-compare -Wstrict-prototypes -Wno-sign-conversion -Wuninitialized")
 set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++11 -Wno-attributes -Wunused-label -Wunused-variable -Wunused-value -Wno-pragmas -Wempty-body -Wshadow -Wparentheses -Wswitch -Wsign-compare -Wno-sign-conversion -Wuninitialized")
 
+# Windows specific flags and variables
+if(WIN32)
+    set(CMAKE_CXX_STANDARD_LIBRARIES "-static-libgcc -static-libstdc++ -lwsock32 -lws2_32 ${CMAKE_CXX_STANDARD_LIBRARIES}")
+    set(CMAKE_EXE_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS} -Wl,-Bstatic,--whole-archive -lwinpthread -Wl,--no-whole-archive")
+    set(GNULIB_NAMESPACE_SOURCES "${PROJECT_SOURCE_DIR}/core/chromosome.cpp" 
```

Comment:
I think I see what you're doing with this `GNULIB_NAMESPACE_SOURCES` business.  It is elegant; the only question I have regards maintenance.  From time to time, I will presumably add something to some source file that will create a dependency on gnulib, and that will break the Windows build until you notice it and fix it.  Is there a reason not to just do the gnulib include and define for *all* source files?  Are there source files where that causes a problem?  Or does it just make the build substantially slower?  If it works OK, it might make our lives simpler down the road.

Reply:
To understand why we should only define this for some sources, it would help to under how `gnulib` works. It works by selectively defining functions that are missing on the target system. This works great in C code but causes a problem in C++ code. This is because if you have a class with a method which happens to have the same name as one of the functions being replaced by `gnulib`, that method gets redefined and you get linker errors. The `GNULIB_NAMESPACE` define works around this by telling `gnulib` to define its replacement functions as `gnulib::function_name`, so it no longer clashes with any C++ class methods. So you put this define in any source that has such a clash. The problem is then, if you use any of the functions defined by `gnulib` in that same source, it won't work anymore because you now have to refer to them as `gnulib::function_name` instead of just `function_name`. So if we put this in all sources, we will likely have to do a lot more replacing of `function_name` with `gnulib::function_name`, and that will be just as annoying if not more annoying to maintain I think, and will lead to having to touch a lot more files potentially. On the other hand it is possible that this situation doesn't come up that often, so it might be worth testing out... I don't know, can we leave it for now? I can try and test alternatives a little down the road once this is all set up? 

In the mean time I can set up a gh actions CI for the Windows build that can warn me if the build gets broken by any update.

Reply:
I see.  OK, your solution seems best.  Thanks for the explanation.